shelley/chain-and-ledger/formal-spec/chain.tex

```diff
@@ -1272,6 +1272,104 @@ \subsection{Chain Transition}
   \label{fig:rules:chain}
 \end{figure}
 
+\clearpage
+
+\subsection{Byron to Shelley Transition}
+\label{sec:byron-to-shelley}
+
+This section describes how to transition the state held by the Byron ledger to the Shelley ledger.
+The Byron ledger state $\CEState$ is defined in \cite{byron_chain_spec}.
+Figure~\ref{fig:functions:to-shelley} defines a function $\fun{toShelley}$
+which takes an initial nonce $\eta$ and the Byron ledger state,
+and creates the Shelley ledger state.
+
+%%
+%% Figure - Byron to Shelley State Transition
+%%
+\begin{figure}[htb]
+  \emph{Applying a reward update}
+  %
+  \begin{align*}
+      & \fun{toShelley} \in \Seed \to \CEState \to \ChainState \\
+      & \fun{toShelley}~\eta~
+      \left(
+        \begin{array}{c}
+          \var{s_{last}} \\
+          \var{sgs} \\
+          \var{h} \\
+          \var{utxo} \\
+          \var{ds} \\
+          \var{us}
+        \end{array}
+      \right)
+      =
+      \left(
+        \begin{array}{c}
+          (\eta,~\eta,~\eta) \\
+          \emptyset \\
+          h \\
+          \var{s_{last}} \\
+          \epoch{s_{last}} \\
+          \\
+          \left(
+            \begin{array}{c}
+              \left(
+                \begin{array}{c}
+                  0 \\
+                  4.5\times 10^{16} - \fun{balance}~{utxo}
+                \end{array}
+              \right) \\
+              \left(
+                \begin{array}{c}
+                  (\emptyset, \emptyset) \\
+                  (\emptyset, \emptyset) \\
+                  (\emptyset, \emptyset) \\
+                  \emptyset \\
+                  \emptyset \\
+                  0
+                \end{array}
+              \right) \\
+              \left(
+                \begin{array}{c}
+                  \left(
+                    \begin{array}{c}
+                      \var{utxo} \\
+                      0 \\
+                      0 \\
+                    \end{array}
+                  \right) \\
+                  \left(
+                    \begin{array}{c}
+                      \emptyset \\
+                      \emptyset \\
+                      \emptyset \\
+                      \emptyset \\
+                    \end{array}
+                  \right) \\
+                  \left(
+                    \begin{array}{c}
+                      \emptyset \\
+                      \emptyset \\
+                      \emptyset \\
+                      \emptyset \\
+                    \end{array}
+                  \right) \\
+                  \var{us} \\
+                \end{array}
+              \right) \\
+            \end{array}
+          \right) \\
+          \\
+          \Nothing \\
+          \emptyset \\
```

Comment:
shouldn't we take the genesis key delegation from `DIState` / `ds` ?

Reply:
yea, I was wondering about that too. Currently the genesis keys are just in the `CHAIN` environment, since they never change. But in order to inherit them, we could move them to the state?